### ISSUE TITLE
Maybe discourage re-declaration of final method

### ISSUE THREAD

**Issue by twoertwein** [PyCQA/flake8-pyi#300]
```py
from typing import final

class A:
   @final
   def a(self) -> None : ...

class B(A):
   def a(self) -> None : ...  # would be nice to error here
```

Can probably only be supported by flake8-pyi, if the super-class is within the same file as the sub-class.

**Comment by JelleZijlstra** [comment]
Type checkers should flag this. Why should we duplicate the check in flake8-pyi?

**Comment by twoertwein** [comment]
I think mypy doesn't catch this in pyi files but it makes sense to let type checkers find this.

**Comment by JelleZijlstra** [comment]
Sounds like a bug in mypy :)

**Comment by AlexWaygood** [comment]
Mypy definitely catches this in pyi files in some situations — we have to use a `type: ignore` here to get around the fact that the `odict_keys` class in the `collections` module inherits from a class that's unsubclassable at the Python level (but is subclassable at the C level): https://github.com/python/typeshed/blob/62a6c3d616d997c1c6dc170a228c57c9f02ad504/stdlib/collections/__init__.pyi#L331

Curious if you can show a repro for the mypy bug :)

**Comment by AlexWaygood** [comment]
If I have the following in a pyi file, it looks like mypy correctly emits an error for `Eggs.bar()`. It doesn't error for `Eggs.baz()`, but that's because it generally doesn't emit any errors for functions that don't have any type annotations:

```python
from typing import final

class Spam:
    @final
    def bar(self) -> None: ...
    @final
    def baz(self): ...


class Eggs(Spam):
    def bar(self) -> None: ...
    def baz(self): ...
```

**Comment by AlexWaygood** [comment]
Anyway as @JelleZijlstra says, this is something for type checkers to worry about -- if they're not emitting errors for this, they should be, and that should be reported as a bug :)
